File: 2473-max-sum-of-a-pair-with-equal-sum-of-digits/2473-max-sum-of-a-pair-with-equal-sum-of-digits.py
class Solution:
    def maximumSum(self, nums: List[int]) -> int:
        maxsum = -1
        dict = { }
        for i in range(0,len(nums)):
            temp1 = 0
            num = nums[i]
            while num>0 :
                temp1 += num%10
                num = num//10
            
            if temp1 in dict:
                sum = dict[temp1]+nums[i]
                maxsum = max(maxsum,sum)
                dict[temp1] = max(dict[temp1],nums[i])

            else:
                dict[temp1]=nums[i]

        return maxsum

File: 2473-max-sum-of-a-pair-with-equal-sum-of-digits/2473-max-sum-of-a-pair-with-equal-sum-of-digits.java
class Solution {
    public int maximumSum(int[] nums) {
        HashMap<Integer,Integer> hm = new HashMap<>();
        int maxsum = -1;
        for(int i=0;i<nums.length;i++){
            int temp1 = 0;
            int num = nums[i];
            while(num > 0){
                int temp = num%10;
                num = num/10;
                temp1 = temp1+temp;
            }
            if(hm.containsKey(temp1)){
                int sum = hm.get(temp1)+nums[i];
                
                maxsum = Math.max(sum,maxsum);
                hm.put(temp1,Math.max(hm.get(temp1),nums[i]));
            }else{
                hm.put(temp1,nums[i]);
            }
        } 
        return maxsum;
    }
}